hedera-node/src/main/java/com/hedera/services/ledger/accounts/staking/RewardCalculator.java

```diff
@@ -73,13 +80,39 @@ boolean isWithinRange(final long stakePeriodStart, final long todayNumber) {
 		return stakePeriodStart > -1 && stakePeriodStart < todayNumber - 1;
 	}
 
-	long computeReward(final MerkleAccount account, final long stakedNode, final long todayNumber) {
+	long computeReward(final MerkleAccount account, final long stakedNode, final long todayNumber,
+			final long stakePeriodStart) {
 		final var stakedNodeAccount = stakingInfo.get().get(EntityNum.fromLong(stakedNode));
 		final var rewardSumHistory = stakedNodeAccount.getRewardSumHistory();
-		final var stakePeriodStart = account.getStakePeriodStart();
+
 		// stakedNode.rewardSumHistory[0] is the reward for all days up to and including the full day todayNumber - 1,
 		// since today is not finished yet.
 		return account.isDeclinedReward() ? 0 :
 				account.getBalance() * (rewardSumHistory[0] - rewardSumHistory[(int) (todayNumber - 1 - (stakePeriodStart - 1))]);
 	}
+
+	public long latestRewardableStakePeriodStart() {
+		return networkCtx.get().areRewardsActivated() ? LocalDate.now(zoneUTC).toEpochDay() - 1 : Long.MIN_VALUE;
+	}
+
+	public long updateRewardChanges(final MerkleAccount account, final Map<AccountProperty, Object> changes) {
+		final var result = computeRewards(account);
```

Comment:
Since this class is already not thread-safe, maybe we can have two fields,
```
long accountReward;
long accountUpdatedStakingPeriod;
```
...and then make `computeRewards()` a `void` method that just sets those two variables; this avoids creating a a `Pair` and auto-boxing two `long`s.